README.md

```diff
@@ -0,0 +1,107 @@
+# tracing-active-tree
+
+`tracing-active-tree` is a Rust library that allows developers to retrieve the current calling stack at runtime on demand. The calling stack is organized as [span](https://docs.rs/tracing/latest/tracing/span/index.html) trees, and can be used for both synchronous and asynchronous contexts.
+
+One common use case for this library is to print or log the calling trees on error for troubleshooting purposes. Unlike capturing stack backtraces using `std::backtrace::Backtrace`, `tracing-active-tree` offers several advantages:
+
+- It supports asynchronous functions.
+- It records trees with additional context attributes.
+
+However, `tracing-active-tree` has a few disadvantages as well:
+
+- Spans need to be declared beforehand.
+- It requires more CPU and memory resources.
+
+Two similar libraries, `async-backtrace` and `await-tree`, are also available. `tracing-active-tree` takes inspiration from their implementation. However, both `async-backtrace` and `await-tree` require declaring spans using methods provided by the library, which can be inconvenient for existing codebases. In contrast, `tracing-active-tree` does not require any changes to the existing code. It only requires registering an extra Layer to the tracing subscribers registry.
+
+## Installation
+
+Add the following line to your `Cargo.toml` file:
+
+```toml
+tracing-active-tree = { git = "https://github.com/tikv/tracing-active-tree.git", branch = "master" }
+```
+
+## Usage
+
+*Run the following example using `cargo run --example simple`.*
+
+1. Import the library:
+
+```rust
+use tracing_active_tree::layer::{self, CurrentStacksLayer};
+```
+
+2. Create spans that you wish to display in the calling trees:
+
+```rust
+#[instrument(fields(answer = 43))]
+async fn foo() {
+    bar().await;
+}
+
```

Comment:
Is it better to indent after an ordered list?
```suggestion
1. Import the library:

	```rust
	use tracing_active_tree::layer::{self, CurrentStacksLayer};
	```

2. Create spans that you wish to display in the calling trees:

	```rust
	#[instrument(fields(answer = 43))]
	async fn foo() {
	    bar().await;
	}

```

Reply:
Fixed, PTAL~

p.s. other warnings from [`markdownlint`](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) were also fixed.